test/test-dot.js

```diff
@@ -55,6 +57,61 @@ test('reply.view with dot engine .dot file', t => {
   })
 })
 
+test('reply.view with dot engine .dot file should create non-existent destination', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  const engine = require('dot')
+  engine.log = false
+
+  fastify.register(require('../index'), {
+    engine: {
+      dot: engine
+    },
+    options: {
+      destination: 'non-existent'
+    }
+  })
+
+  t.tearDown(() => rmdirSync('non-existent'))
+
+  fastify.get('/', (req, reply) => {
+    reply.view('testdot')
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+    t.true(existsSync('non-existent'))
+    fastify.close()
+  })
+})
+
+test('reply.view with dot engine .dot file should log WARN if template not found', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  const engine = require('dot')
+  engine.log = false
+
+  t.tearDown(() => rmdirSync('empty'))
+
+  fastify.log.warn = function mockWarn (message) {
```

Comment:
As an improvement, you may change this code to test the logger with this one:

https://github.com/fastify/fastify/blob/61c8a7bc3c91907e97813b631d35ae1f250b40e2/test/logger.test.js#L1099